### PR TITLE
updated .hpp files 

### LINKED ISSUE #9: Complete documentation
#### Description of the problem

Documentation for some classes and functions isn't complete in `.hpp` files. The issue can be resolved by adding docstrings to all the missing places.

### PR DESCRIPTION
**Pull request by akankshasehrawat**
#### References to other Issues or PRs or Relevant literature
 #9 

#### Other comments

### CODE REVIEW

File: adaboost/cuda/cuda_data_structures.hpp [hunk 1]
```diff
@@ -7,6 +7,7 @@ namespace adaboost
 {
     namespace cuda
     {
+
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures.hpp]:
```suggestion
```

File: adaboost/cuda/cuda_data_structures.hpp [hunk 2]
```diff
@@ -104,13 +105,25 @@ namespace adaboost
                     */
                     ~VectorGPU();
             };
+            
+            /*
+            * This class is represented 
+            * for storing data in a vector
+            * and taking their dot product.
+            */
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures.hpp:112]: It's not a class it's a function.

File: adaboost/cuda/cuda_data_structures.hpp [hunk 3]
```diff
@@ -124,25 +137,52 @@ namespace adaboost
...
+                              
+                              
+                    /*
+                    * Copies the data from GPU to CPU.
+                    */
                     void copy_to_host();
-
+                    
+                    /*
+                    * Copies the  data from CPU to GPU.
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures.hpp]:
```suggestion
                    * Copies the data from CPU to GPU.
```

File: adaboost/cuda/cuda_data_structures.hpp [hunk 4]
```diff
@@ -124,25 +137,52 @@ namespace adaboost
...
+                    */
                     void copy_to_device();
-
+                    
+                    /*
+                    * Returns the size of the row
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures.hpp]:
```suggestion
                    * Returns number of rows
```

File: adaboost/cuda/cuda_data_structures.hpp [hunk 5]
```diff
@@ -104,13 +105,25 @@ namespace adaboost
...
+            */
 
             template <class data_type_vector>
             void product_gpu(const VectorGPU<data_type_vector>& vec1,
                              const VectorGPU<data_type_vector>& vec2,
                              data_type_vector& result,
                              unsigned block_size=0);
-
+            /*
+            * This class represents matrices used for storing data
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures.hpp:121]:
```suggestion
            * This class represents matrices used for storing data on GPU
```

File: adaboost/cuda/cuda_data_structures.hpp [hunk 6]
```diff
@@ -124,25 +137,52 @@ namespace adaboost
...
+                    /*
+                    * Copies the  data from CPU to GPU.
+                    */
                     void copy_to_device();
-
+                    
+                    /*
+                    * Returns the size of the row
+                    * By default returns the size of the
+                    * vector row on GPU.
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures.hpp]:
```suggestion
                    * By default returns number of rows on GPU.
```

File: adaboost/cuda/cuda_data_structures_impl.hpp [hunk 7]
```diff
@@ -16,11 +16,19 @@ namespace adaboost
...
+            */
             template <class data_type_vector>
             data_type_vector*
             VectorGPU<data_type_vector>::
             _reserve_space_gpu(unsigned _size_gpu)
             {
+                /*
+                * For reserving space in GPU memory accoring to a given size.
+                * A new pointer is retured.
+                */
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures_impl.hpp]:
```suggestion
```

File: adaboost/cuda/cuda_data_structures_impl.hpp [hunk 8]
```diff
@@ -16,11 +16,19 @@ namespace adaboost
     {
         namespace core
         {
+            /*
+            *This class represents the GPU version
+            * of adaboost::core::Vector.
+            */
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures_impl.hpp]:
```suggestion
```

File: adaboost/cuda/cuda_data_structures.hpp [hunk 9]
```diff
@@ -104,13 +105,25 @@ namespace adaboost
...
                              data_type_vector& result,
                              unsigned block_size=0);
-
+            /*
+            * This class represents matrices used for storing data
+            * in two dimensions, implemented using C++ arrays.
+            *
+            * @tparam data_type_matrix Data type of the elements
+            *     supported by C++.
+            */
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures.hpp]:
```suggestion
            * This class represents GPU version of adaboost::core::Matrix.
            *
            * @tparam data_type_matrix Data type of the elements
            *     supported by C++.
            */
```

File: adaboost/cuda/cuda_data_structures.hpp (hunk 2 above)
Comment (akankshasehrawat) [reply adaboost/cuda/cuda_data_structures.hpp:112]: okay!

File: adaboost/cuda/cuda_data_structures.hpp [hunk 10]
```diff
@@ -104,13 +104,24 @@ namespace adaboost
                     */
                     ~VectorGPU();
             };
+            
+            /*
+            * This function is represented 
+            * for storing data in a vector
+            * and taking their dot product.
+            */
```
Comment (czgdp1807) [inline adaboost/cuda/cuda_data_structures.hpp]:
```suggestion
            /*
            * This function computes
            * dot product of two vectors on
            * GPU.
            */
```

**Pull request merged**